hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMMultiTenantManagerImpl.java

```diff
@@ -690,9 +686,11 @@ public boolean isUserAccessIdPrincipalOrTenantAdmin(String accessId,
   @Override
   public boolean isTenantEmpty(String tenantId) throws IOException {
 
-    // TODO: Do we need to iterate cache here as well? Very cumbersome if so.
-    //  This helper function is a placeholder for the isTenantEmpty check,
-    //  once tenantCache/Ranger is fixed this will be removed.
+    if (tenantCache.containsKey(tenantId) &&
+        !tenantCache.get(tenantId).isTenantEmpty()) {
```

Comment:
The `tenantCache` is supposed to have all the user data for each tenant. I don't think we need to iterate the DB as well. If the tenant is not in the cache, we can throw an IOException like `listUsersInTenant` does, since this would indicate a bug in our OM code.

Reply:
The caller of `isTenantEmpty` already throws an `OMException` when we return false. Should we explicitly throw an `IOException` here as well? cc: @errose28

Reply:
I think so, because they are two different errors. If this method returns false and the caller throws OMException, it is because the client tried to delete a non-empty tenant. If this method is called with a tenant that does not exist, there is a bug in our code and neither the true or false responses make sense.

This assumes the caller already checked the validity of the tenant it got from the client though. If this method is expected to do that as well, we can through an OMException with TENANT_NOT_FOUND return code from this method instead.